### PR TITLE
Use TypeScript

### PR DESCRIPTION
**Pull request by Anduin2017**
* Use TypeScript.
* Remove render.js
* Add lint for style
* Add document

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -4,6 +4,44 @@
   "lockfileVersion": 1,
   "requires": true,
   "dependencies": {
+    "@angular/compiler": {
```
Comment (liuxintong) [inline package-lock.json:7]: Please clean up unnecessary dependencies, I don't think we rely on angular.

**Review by liuxintong — COMMENTED** [review]
Overall looks good, and most issues are about coding styles. I'm not an expert in frontend development, not sure if the case of kinds of names are aligning with the common habits. In addition, please double check the dependencies, it would be great to keep it neat.

File: src/ConfigStore.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { remote } from 'electron';
+
+export class ConfigStore {
+
```
[the remark this answers is not in the document]
Comment (Anduin2017) [reply src/ConfigStore.ts:4]: Before a method, there always shall be a blank line.

File: src/app.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
+import { hostname } from 'os';
+import Axios from 'axios';
```
[the remark this answers is not in the document]
Comment (Anduin2017) [reply src/app.ts:2]: No. You can't decide what a library exports.

File: src/app.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+      }
+    });
+  }
+
+  private init() {
+    (document.getElementById('server') as HTMLInputElement).value = ConfigStore.get('server');
+    (document.getElementById('account') as HTMLInputElement).value = ConfigStore.get('account');
+
+    if (!ConfigStore.get('hostid')) {
+      ConfigStore.set('hostid', v4());
```
[the remark this answers is not in the document]
Comment (Anduin2017) [reply src/app.ts:32]: The uuid is a js library which doesn't typed intelligent annotations so you can't rename it or the runtime can't find the library.

File: package-lock.json (hunk 1 above)
Comment (Anduin2017) [reply package-lock.json:7]: No we don't. This file is auto generated. @angular/compiler is referenced by codelyzer which is necessary for lint.

File: index.html [hunk 5]
```diff
@@ -1,19 +1,27 @@
...
+<head>
+  <meta charset="UTF-8">
+  <!-- https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP -->
+  <meta http-equiv="Content-Security-Policy" content="default-src 'self' *; script-src 'self'">
+  <meta http-equiv="X-Content-Security-Policy" content="default-src 'self' *; script-src 'self'">
+  <title>xConnect</title>
+</head>
+
+<body>
+  <form id="connect-form">
```
[the remark this answers is not in the document]
Comment (Anduin2017) [reply index.html:13]: Using a form is the best practice for this case. I will explain you that later.

**Comment by Anduin2017** [comment]
@liuxintong Fixed. Please review again.

**Comment by liuxintong** [comment]
Thanks for your contribution on moving the project to TypeScript. :)

**Pull request merged**